### PR TITLE
Add --force to `kubectl update`

### PR DESCRIPTION
**Pull request by caesarxuchao**
Add flag `--force` to `kubectl update`. 

The intended use is `kubectl update --force -f example.json`
When `--force` is specified, the resource specified in example.json is deleted and then re-created.
The force update accepts flags accepted by `kubectl delete`, such as `cascade`, except that the `ignore-not-found` flag is hardcoded to 'true', this means if the resource does not already exist, the resource will be created.

ref #10202 #8683
@bgrant0607 @jlowdermilk @nikhiljindal @krousey

### CODE REVIEW

**Comment by mbforbes** [comment]
@ixdy any idea why Jenkins doesn't automatically start on PRs like these?

**Comment by mbforbes** [comment]
@bgrant0607 has a ton of PRs assigned so I'm loath to assign, but he's the only TL mentioned. Brian, please feel free to toss to another TL.

**Comment by ixdy** [comment]
@mbforbes the Jenkins pull request plugin is broken and is using unauthenticated API requests, which have a low ratelimit, and so is unable to update any PRs.

See jenkinsci/ghprb-plugin#117 for details.

**Comment by mbforbes** [comment]
@ixdy thanks for the details. I'll just verify manually for now.

**Comment by mbforbes** [comment]
Jenkins build/test passed for this PR.

@caesarxuchao could you please check/kick shippable?

**Comment by bgrant0607** [comment]
@jlowdermilk Please do a review. Syntax/semantics looks like what I expected.

File: pkg/kubectl/cmd/update.go [hunk 1]
```diff
@@ -60,6 +63,10 @@ func NewCmdUpdate(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmd.MarkFlagRequired("filename")
 	cmd.Flags().String("patch", "", "A JSON document to override the existing resource. The resource is downloaded, patched with the JSON, then updated.")
 	cmd.MarkFlagRequired("patch")
+	cmd.Flags().Bool("force", false, "Delete and re-create the specified resource")
+	cmd.Flags().Bool("cascade", true, "Only relevant during a force update. If true, cascade the delete of resources managed by this resource (e.g. Pods created by a ReplicationController).  Default true.")
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/update.go]: I feel like this flag should prompt a warning. We tell users to use `rolling-update` for destructive updates on replicated pods. If we also allow them to update rcs with `kubectl update --force` we probably want to make sure they actually did want to delete all replicas at once and recreate them.

**Comment by jlowdermilk** [comment]
LGTM, with one suggestion: I would like to see a warning when cascade=true, especially if that's the default. If the user has a good reason for doing a delete/recreate on all pods in an rc at once, they can ignore the warning.

**Comment by caesarxuchao** [comment]
Pushed a fix for Shippable. Will squash before merge.

**Comment by bgrant0607** [comment]
A warning makes sense to me.

**Comment by caesarxuchao** [comment]
I added the warning, I also changed cascade default to false.

File: pkg/kubectl/cmd/update.go [hunk 2]
```diff
@@ -160,3 +176,76 @@ func updateWithPatch(cmd *cobra.Command, args []string, f *cmdutil.Factory, patc
...
+		Do()
+	err = r.Err()
+	if err != nil {
+		return err
+	}
+	//Update will create a resource if it doesn't exist already, so ignore not found error
+	ignoreNotFound := true
+	// By default use a reaper to delete all related resources.
+	if cmdutil.GetFlagBool(cmd, "cascade") {
+		glog.Warningf("\"cascade\" is set, kubectl will cascade the deletion of the resources managed by this resource (e.g. Pods created by a ReplicationController). Consider using \"kubectl rolling-update\" if you really plan to update a ReplicationController together with its Pods.")
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/update.go]: Wording nits for simplicity/clarity:
- s/will cascade the deletion of the/will delete and recreate all
- s/if you really plan/if you want

**Comment by caesarxuchao** [comment]
Done. Thanks.

**Comment by jlowdermilk** [comment]
LGTM, just needs squash.

**Comment by caesarxuchao** [comment]
Squashed.

**Comment by mbforbes** [comment]
All checks are green! Now we just need 2 lead LGTMS.

@bgrant0607 do you want to be one and rope in a second?

**Comment by bgrant0607** [comment]
LGTM

**Comment by bgrant0607** [comment]
ok to merge

**Comment by mbforbes** [comment]
Ugh, I hate to be pedantic, but I'm supposed to wait from another LGTM from another lead. Anyone else that we'd like to take a look?

**Comment by bgrant0607** [comment]
Oh, I thought it was just 2 LGTMs -- a regular one and one from a lead. 

@quinton-hoole, could I please get another LGTM?

**Comment by quinton-hoole** [comment]
I didn't go through it in too much detail, but the code and motivation for merging seems reasonable.  LGTM. ok-to-merge.

**Pull request merged**